src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/DefinitionEndpoint.cs

```diff
@@ -201,4 +231,77 @@ public RegistrationExtensionResult GetRegistration(VSInternalClientCapabilities
             _ => null
         };
     }
+
+    private async Task<Range> GetNavigateRangeAsync(IDocumentSnapshot documentSnapshot, BoundAttributeDescriptor? attributeDescriptor, ILogger logger, CancellationToken cancellationToken)
+    {
+        if (attributeDescriptor is not null)
+        {
+            logger.LogInformation("Attempting to get definition from an attribute directly.");
+
+            var originCodeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+            var range = await TryGetPropertyRangeAsync(originCodeDocument, attributeDescriptor.GetPropertyName(), _documentMappingService, logger, cancellationToken).ConfigureAwait(false);
+
+            if (range is not null)
+            {
+                return range;
+            }
+        }
+
+        // When navigating from a start or end tag, we just take the user to the top of the file.
+        // If we were trying to navigate to a property, and we couldn't find it, we can at least take
+        // them to the file for the component. If the property was defined in a partial class they can
+        // at least then press F7 to go there.
+        return new Range { Start = new Position(0, 0), End = new Position(0, 0) };
```

Comment:
I'm surprised we don't have a `Range.Zero`

Reply:
Sadly, its not our type, but maybe a future C# version will allow extension everything :D